app/src/main/java/pl/pwr/zpi/bcycle/mobile/TripCreationMapActivity.kt

```diff
@@ -0,0 +1,326 @@
+        val mapSettings = map?.uiSettings
+        mapSettings?.isZoomGesturesEnabled = true
+        mapSettings?.isScrollGesturesEnabled = true
+    }
+
+    // region markerClickListener
+    override fun onMarkerClick(marker: Marker?): Boolean {
+        val array = arrayListOf<String>(
+            getString(R.string.set_as_start_point), getString(R.string.set_as_end_point), getString(
+                R.string.remove_it
+            )
+        )
+        LovelyChoiceDialog(this)
+            .setTopColorRes(R.color.violet)
+            .setTitle(resources.getString(R.string.prompt_marker_what_to_do))
+            .setIcon(R.drawable.bike_icon)
+            .setItemsMultiChoice(
+                array
+            ) { _: List<Int>, items: List<String> ->
+                if (items.contains(array[0], array[2]) || items.contains(array[1], array[2])) {
+                    showToastError((R.string.warning_cant_set_and_delete_marker))
+                } else if (items.contains(array[0]) && items.contains(array[1])) {
+                    markerStartPoint?.setIcon(null)
+                    markerFinishPoint?.setIcon(null)
+                    markerStartPoint = marker
+                    markerFinishPoint = marker
+                    marker?.setIcon(BitmapDescriptorFactory.fromResource(R.drawable.icon_bike_black))
+                    marker?.title = getString(R.string.start_and_end_point)
+                    marker?.showInfoWindow()
+                } else if (items.contains(array[0])) {
+                    markerStartPoint?.setIcon(null)
+                    marker?.setIcon(BitmapDescriptorFactory.fromResource(R.drawable.icon_bike_black))
+                    markerStartPoint = marker
+                    marker?.title = getString(R.string.start_point)
+                    marker?.showInfoWindow()
+                } else if (items.contains(array[1])) {
+                    markerFinishPoint?.setIcon(null)
+                    marker?.setIcon(BitmapDescriptorFactory.fromResource(R.drawable.icon_bike_black))
+                    markerFinishPoint = marker
+                    marker?.title = getString(R.string.end_point)
+                    marker?.showInfoWindow()
+                } else if (items.contains(array[2])) {
+                    if (marker == markerStartPoint) markerStartPoint = null
+                    if (marker == markerFinishPoint) markerFinishPoint = null
+                    marker?.remove()
+                }
+            }
+            .setConfirmButtonText(R.string.confirm)
+            .show()
+        return true
+    }
+
+    // endregion markerClickListener
+
+    // region requesting permission region
+
+    override fun onLocationChanged(location: Location) {
+        val latLng = LatLng(location.latitude, location.longitude)
+        val cameraUpdate = CameraUpdateFactory.newLatLngZoom(latLng, 12f)
+        map?.animateCamera(cameraUpdate)
+    }
+
+    private fun requestPermission(
```

Comment:
`RecordTripActivity` also does that, perhaps we could move it to a different file (`LocationActivity` superclass, or a class prepared to handle it that we could just call here and in the response method with callbacks)